### PR TITLE
prepare_release.py: create advisory live ids directly

### PR DESCRIPTION
**Pull request by sosiouxme**
we now have the ability to create our own live ids, so let's do that instead of asking docs.
UPDATE: now relies on [elliott creating live id](https://github.com/openshift/elliott/commit/9e4a6b3efb7eda5c9c4734d610d5644ca64cd298).

### CODE REVIEW

File: pyartcd/pyartcd/pipelines/prepare_release.py [hunk 1]
```diff
@@ -650,24 +650,13 @@ async def verify_attached_operators(self, *advisories: List[int]):
...
-{main_advisory}: https://errata.devel.redhat.com/advisory/{advisories[main_advisory]}
-
-This is the current set of advisories we intend to ship:
-"""
+    async def create_live_ids(self, advisories: Dict[str, int]):
         for kind, advisory in advisories.items():
-            content += (
-                f"- {kind}: https://errata.devel.redhat.com/advisory/{advisory}\n"
+            _LOGGER.info(f"Giving {kind} advisory {advisory} a Live ID.")
+            await exectools.cmd_assert_async(
```
Comment (joepvd) [inline pyartcd/pyartcd/pipelines/prepare_release.py:656]: How does this return when a LiveID already has been assigned?

In order to have meaningful return codes, we might need to read if a LiveID has been assigned already, like this: https://github.com/openshift/aos-cd-jobs/blob/76d7ee16bb34dd81eb284ac5aaac53cd1ca9533b/pipeline-scripts/release.groovy#L120

**Review by joepvd — COMMENTED** [review]
~This meeting could have been an email~
This email could have been an API call

**Comment by joepvd** [comment]
Also, once an official API exists, we should use that: https://issues.redhat.com/browse/CLOUDWF-6578

File: pyartcd/pyartcd/pipelines/prepare_release.py [hunk 2]
```diff
@@ -650,24 +650,13 @@ async def verify_attached_operators(self, *advisories: List[int]):
...
+                f"curl -X POST --negotiate -u : https://errata.devel.redhat.com/errata/set_live_advisory_name/{advisory}\n"
```
Comment (vfreex) [inline pyartcd/pyartcd/pipelines/prepare_release.py:657]: Would like to add `--fail` option to return non-zero code if response status is not 200-399.

File: pyartcd/pyartcd/pipelines/prepare_release.py (hunk 1 above)
Comment (sosiouxme) [reply pyartcd/pyartcd/pipelines/prepare_release.py:656]: it turns out to be idempotent AFAICS. checking the return code and maybe retrying would probably be a good idea though (and a slack msg if it fails - it's not critical). if it's any more complicated than this command (or even if that's all it is), might make a good elliott command

Comment (vfreex) [reply pyartcd/pyartcd/pipelines/prepare_release.py:656]: I think we only need to request the live ID just after a new advisory is created. When updating an existing advisory, we don't need to request again.

Comment (vfreex) [reply pyartcd/pyartcd/pipelines/prepare_release.py:656]: So this function can be chained after the create advisory logic.

Comment (sosiouxme) [reply pyartcd/pyartcd/pipelines/prepare_release.py:656]: yeah this should just go in `elliott create`

File: pyartcd/pyartcd/pipelines/prepare_release.py (hunk 2 above)
Comment (sosiouxme) [reply pyartcd/pyartcd/pipelines/prepare_release.py]: did that in [elliott](https://github.com/openshift/elliott/commit/9e4a6b3efb7eda5c9c4734d610d5644ca64cd298)

**Comment by sosiouxme** [comment]
I haven't run this but I think it's safe.

**Pull request merged**